File: csrgen.java
package csrgen;
import java.util.Scanner;

public class Zertifikat {
/*
1) Reading the Name of the Domain the Cert is being made for
*/
    String FOLDER = "~/Zertifikate/";
    String KUERZEL = "DWE";
    int TYPE;
    String DOMAIN;
    String DIRECTORY;
    Scanner read = new Scanner(System.in);

    public void domainEingeben(){
        System.out.print("Domain: ");
        DOMAIN = read.nextLine();

    }
/*
2) Choosing Cert Type and confirming
*/
    public void typeEingeben(){
        System.out.prinf("%nNow let's choose the certificate type.%nThese are the Options:%n1.) AlphaSSL%n2.) Wildcard%n3.) SAN (Mit oder ohne EV)%n4.) LetsEncrypt%n5.) GeoTrust EV");
        TYPE = read.nextInt();
        String PREFIX;
        String CN;
        switch(TYPE){
            case 1:    //AlphaSSL
            case 5:    //GeotrustEV                     
                PREFIX=("www.");              
            case 2:    //Wildcard
                PREFIX=("wc.");
                CN=("*.");
                System.out.println("EV? yes/no: ");
                String EV;
                EV = read.nextLine();
            case 3:     //SAN
                PREFIX=("san.");
            case 4:     //LetsEncrypt
                System.out.println("LetsEncrypt Zertifikate werden über service2 ausgestellt:");
                System.out.println("https://service2.continum.net/services/ssl-certificates");
        
            default:
                System.out.println("Number not bet 1 and 5.");
        }
/*
3) Creating dedicated Folder an executing the keygen-commands in it
*/

    public void datenBestätigen() {
        System.out.println("Typ des Zertifikates: "+TYPE);
    } 
}

File: csrgen.py
#!/usr/bin/env python3
from logging import exception
import sys # sys.stdout.write('Hello World')
import datetime
from pathlib import Path
from time import sleep
import subprocess  #process = subprocess.check_output(['command1', 'command2'])
import re

AlphaSSL = 1
Wildcard = 2
SAN = 3
LetsEncrypt = 4
GeoTrust = 5
SUBDOMAIN = False
CN = "placeholder"
PREFIX = "placeholder"
DOMAIN_UNCONVERTED = "placeholder"
DOMAIN = "placeholder"
#global LAND
#global STADT
#global BUNDESLAND
#global FIRMENNAME
#global ABTEILUNGSNAME
#global FQDN

#yes or no optional functions
def askForYesOrNo(question):
    reply = str(input(question+' (y/n): ')).lower().strip()
    if reply[0] == 'y': return True
    if reply[0] == 'n': return False
    else: return askForYesOrNo("Uhhhh... please enter ")

def askForSubdomain():
    global SUBDOMAIN
    reply = askForYesOrNo('Hat die Domain eine Subdomain? (Nicht www.)')
    if reply == True: print('Format:   subdomain.domain.tld');    SUBDOMAIN=True
    else:             print('Format: (Nicht www.)domain.tld');  SUBDOMAIN=False

def askForCertType():
    try:
        CERTIFICATE_TYPE = int(input("""
        ᵀʰᵃⁿᵏ ʸᵒᵘ ᶠᵒʳ ᵘˢⁱⁿᵍ, 𝘾𝙎𝙍𝙜𝙚𝙣
        Wähle den Zertifikatstyp
           AlphaSSL          = 1
           Wildcard          = 2
           SAN               = 3
           LetsEncrypt       = 4
           GeoTrust EV       = 5
                             ↳ """))
        return CERTIFICATE_TYPE
    except: print('Das ist keine Nummer zwischen 1-5, versuchs nochmal.')

def setPrefixAndCN(CERTIFICATE_TYPE):
    global SUBDOMAIN
    match CERTIFICATE_TYPE:
        case 1 | 5:
            if CERTIFICATE_TYPE == 5: EV="yes"
            #If it has a subdomain the prefix is not needed
            if SUBDOMAIN == True: PREFIX=""
            else: PREFIX="www."; CN="www."
        case 2:
            PREFIX="wc."; CN="*."
            # Ask if its a certificate with EV
            print(' , , Ist es ein Zertifikat mit Extended Validation (EV)?')
            EV=askForYesOrNo()
            askForSubdomain()
        case 3:
            PREFIX="san."
        case 4:
            # prompts to the SB2-Website and then exits
            SB2LINK='https://service.continum.net/services/ssl-certificates'
            print('LetsEncrypt-Zertifikate kannst du über den Service2 bestellen:' +SB2LINK)
        case _:
            print('Das ist keine Nummer zwischen 1-5, versuchs nochmal.')
            sleep(2); askForCertType()

#Converts umlauts with idn
def convertUmlauts():
    global DOMAIN_UNCONVERTED
    convertDomain = subprocess.run(['idn', DOMAIN_UNCONVERTED], stdout=subprocess.PIPE) #Doesnt have error handling yet
    Umlaute = re.search(pattern="*[äÄöÖüÜ]*")
    match DOMAIN_UNCONVERTED:
        case Umlaute: DOMAIN=convertDomain
            # 127 is the exitcode($?) in case a command is not installed. In this case csrgen will exit.
            if $? == 127: print('Installiere idn um Umlaute zu konvertieren'); exit 127
        case _: DOMAIN=DOMAIN_UNCONVERTED

#def createAndPrint():
#    generateCSR()
#    createFilesAndDirectorys()
#    printInputs()
#    printOutputs()
#    createNotes()
#
#if __name__ == '__main__':
#    askForCertType()
#    askForSubdomain()
#    setPrefixAndCN()
#    askForDomainname()
#    createAndPrint()
#    openOptionals()